src/application/services/useNote.ts

```diff
@@ -166,12 +166,16 @@ export default function (options: UseNoteComposableOptions): UseNoteComposableSt
     /**
      * @todo try-catch domain errors
      */
-    const response = await noteService.getNoteById(id);
-
-    note.value = response.note;
-    canEdit.value = response.accessRights.canEdit;
-    noteTools.value = response.tools;
-    parentNote.value = response.parentNote;
+    try {
+      const response = await noteService.getNoteById(id);
+
+      note.value = response.note;
+      canEdit.value = response.accessRights.canEdit;
+      noteTools.value = response.tools;
+      parentNote.value = response.parentNote;
+    } catch (error) {
+      void router.push('/error');
```

Comment:
I'd suggest to separate 404 errors and other errors. If 4xx error happen, show 404 error page. In case of other errors, show 500 "Unexpected error" page

Reply:
request you to review new changes